### ISSUE TITLE
Strange logic in easyblock, make_dir

### ISSUE THREAD

**Issue by akesandgren** [hpcugent/easybuild-framework#2023]
Isn't the order in make_dir wrong?

if self.cfg['keeppreviousinstall']:
elif clean:
elif build_option('module_only'):
else:

I would expect the build_option('module_only') to come before clean?? Or it may clean out the dir during a --module-only run...

**Comment by boegel** [comment]
@akesandgren we could reverse the logic, but it won't matter much, since the steps that call `make_dir` are skipped under `--module-only`

Are you seeing the install directory being cleaned up under `--module-only` somewhere?

**Comment by akesandgren** [comment]
No, i was just reading the code looking for clues regarding other matters and saw the possibly flawed logic.

**Comment by boegel** [comment]
Feel free to open a PR to fix the logic, the tests should tell us if there are any unexpected side-effects.
